File: server/src/main/java/model/Entry.java
package model;

import java.time.LocalDate;

class Entry {
  private int id;
  private String title;
  private float value;
  private LocalDate date;
  private LocalDate updatedAt;
  private LocalDate createdAt; // server-only
  private int userId;
  private int categoryId; // optional
  private int recurrentId; // optional
}

File: client/src/schemas/entry.ts
import { z } from "zod";

export const entrySchema = z.object({
  id: z.number(),
  title: z.string(),
  value: z.number(),
  date: z.number(),
  updatedAt: z.number(),

  // Relations
  walletId: z.number(),
  categoryId: z.number().optional(),
  recurrentId: z.number().optional(),
});

export type Entry = z.infer<typeof entrySchema>;
